client/proposer/service.go

```diff
@@ -83,20 +83,35 @@ func (p *Proposer) proposeCollations() {
 	feed := p.p2p.Feed(pb.Transaction{})
 	ch := make(chan p2p.Message, 20)
 	sub := feed.Subscribe(ch)
+	collation := []*gethTypes.Transaction{}
+	sizeOfCollation := int64(0)
+
 	defer sub.Unsubscribe()
 	defer close(ch)
 	for {
 		select {
 		case msg := <-ch:
+			log.Info("tx received")
 			tx, ok := msg.Data.(*pb.Transaction)
 			if !ok {
 				log.Error("Received incorrect p2p message. Wanted a transaction broadcast message")
 				break
 			}
 			// log.Debugf("Received transaction: %x", tx)
-			if err := p.createCollation(p.ctx, []*gethTypes.Transaction{legacyutil.TransformTransaction(tx)}); err != nil {
-				log.Errorf("Create collation failed: %v", err)
+			gethtx := legacyutil.TransformTransaction(tx)
+
+			if (sizeOfCollation + int64(gethtx.Size())) > types.CollationSizelimit {
```

Comment:
I'm not sure if this will work. I don't think we are not considering the `indicatorByte` but only the `dataByte`. `CollationSizelimit` is checked after we serialize and pad in first byte as `indicatorByte` right?

Reply:
Ahh , yeah missed that part. But then in order to carry out the check I would have to serialize the transactions and check the size of the blob. If we have to do it every time a transaction is received , it would be very expensive just to make sure the collation is in the size limit.

Reply:
I agree. Since this will be replaced by a `tx_pool`let's not spend too much time. Can we please make sure `(sizeOfCollation + int64(gethtx.Size())) > types.CollationSizelimit` doesn't fail e2e? (i.e.: try running a proposer node and a simulator node)